### PR TITLE
Log rows: Refactor as functional components and remove render preview

### PR DESCRIPTION
**Pull request by matyax**
This PR updates the LogRows and LogRow components.

Notable changes:

- Migrated from class to functional components.
- Removed the preview rendering implementation that rendered 100 lines and all the lines after 2 seconds. There were no reasons found to keep this behavior. In replacement, the preview implementation was updated to fully render a portion of the logs, and a preview version of the rest without any event listeners, styles, and derived values. The rest of the rows will not be fully rendered until the user interacts with the panel, when the remaining rows will be rendered.

Here's a comparison between the timeout render vs updated preview in a dashboard with 6 panels of 4000 logs:

Before:

https://github.com/user-attachments/assets/a0786234-19d9-47ee-be31-442fc0835199

After:

https://github.com/user-attachments/assets/55ace212-c5c4-4477-b987-7000346b1490

The reasons for rendering a preview (with displayed fields or the log entry) and not render empty cells or nothing is to 1) keep the scroll height accurate and 2) allow users to cmd+f log lines.

Other changes:

- Reviewed prop memoization, adding missing memoization and removing unnecessary memoizations.
- Updated checkLogsError / checkLogsSampled to return a message or undefined, as they were producing unnecessary new objects.
- Refactored some props that triggerd unnecessary re-renders.

Not in this PR: tested moving onClick/onMouseEnter/onMouseLeave event listeners from LogRow to LogRows with no significant improvements in any measurable metric that justified the implementation change.

Part of https://github.com/grafana/grafana/issues/98020
Part of https://github.com/grafana/grafana/pull/85781

### CODE REVIEW

File: public/app/features/explore/Logs/Logs.tsx [hunk 1]
```diff
@@ -433,37 +436,26 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
...
-            top: scrollElement.scrollTop + element.getBoundingClientRect().top - window.innerHeight / 2,
+            top: logsContainerRef.current.scrollTop + element.getBoundingClientRect().top - window.innerHeight / 2,
           });
         }
-      });
+
+        return;
+      }
+
+      if (props.scrollElement) {
```
Comment (gtk-grafana) [inline public/app/features/explore/Logs/Logs.tsx:453]: optional nit: define scrollElement in props destructuring and remove the `props.`

File: public/app/features/logs/components/LogRow.tsx [hunk 2]
```diff
@@ -26,7 +23,7 @@ import { LogRowMessage } from './LogRowMessage';
 import { LogRowMessageDisplayedFields } from './LogRowMessageDisplayedFields';
 import { getLogLevelStyles, LogRowStyles } from './getLogRowStyles';
 
-interface Props extends Themeable2 {
+export interface Props {
```
Comment (gtk-grafana) [inline public/app/features/logs/components/LogRow.tsx:26]: if we gonna export this it should have a better name?

Comment (matyax) [reply public/app/features/logs/components/LogRow.tsx:26]: It's usually exported as Props. https://github.com/search?q=repo%3Agrafana%2Fgrafana+%22export+interface+Props%22&type=code

**Comment by matyax** [comment]
The issue with opening the Logs Volume panel has been resolved.

**Review by gtk-grafana — APPROVED** [review]
Everything seems to work as expected, I'm a little worried about the preview functionality change getting mixed up with this PR in case we need to revert one or the other, but not enough to not say ship it! :P 

Great work as always, great to see some of the log panel debt get paid down.

**Comment by matyax** [comment]
Yeah, that's a good point. Unfortunately I took it as part of these changes. Hopefully the components would not have to be reverted to class, but fixed, and if the preview causes troubles we can also back off from that.

Thanks for the review!

**Pull request merged**